lib/pbench/server/api/resources/query_apis/metadata_index/__init__.py

```diff
@@ -91,6 +91,12 @@ def get_index(self, dataset: Dataset, root_index_name: AnyStr) -> AnyStr:
             self.logger.error(f"Indices from metadata table not found {e!r}")
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
+        if index_map is None:
+            self.logger.error(
+                f"server.index-map not found in Metadata for a dataset {dataset!r}"
```

Comment:
I'm not sure you really want the `repr` of the dataset in this case; but, yeah, this extra check is a good catch